tests/unit/cache.test.ts

```diff
@@ -0,0 +1,24 @@
+import { assert } from "deno/testing/asserts.ts"
+import { useDownload } from "hooks"
+import { sandbox } from "../utils.ts"
+
+Deno.test("etag-mtime-check",async () => {
+  await sandbox(async tmpdir => {
+    const src = new URL("https://dist.tea.xyz/ijg.org/versions.txt")
+    await useDownload().download({src, dst: tmpdir.join("test.tar.xz")})
+
+    const mtimePath = await useDownload().hash_key(src).join("mtime")
+    const etagPath = await useDownload().hash_key(src).join("etag")
```

Comment:
so we're testing an implementation detail of how useDownload() works?

Basically this test verifies useDownload() but it doesn't really test that etags work.